### PR TITLE
Remove optional block scrolling

### PR DESCRIPTION
**Pull request by janfaracik**
Removes the scrolling that can sometimes occur when interacting with an optional block component (the checkbox).  As displayed [here](https://github.com/jenkinsci/jenkins/pull/7173#pullrequestreview-1135958641) this can cause strange scrolling jitters when interacting with the page.

### Testing done

Manual testing done on the project configuration screen. Open a project, click Configure, click the 'Discard old builds' checkbox at the top of the page, notice the page no longer jumps down.

### Proposed changelog entries

- N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@jenkinsci/sig-ux 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7278"></a>

### CODE REVIEW

**Comment by timja** [comment]
> Testing done
> Manual

@janfaracik that section should include what testing you've done specifically, e.g. what page / how you tested it, manual isn't very useful.

**Review by timja — APPROVED** [review]
Much better, as an example of where this is bad, just toggle discard old builds on master with it near the top of the screen it will scroll quite far off screen:

image

**Review by NotMyFault — APPROVED** [review]

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (timja) [reply war/src/main/webapp/scripts/hudson-behavior.js:1925]: Yes I just tested in browser console:

```javascript
function updateOptionalBlock(value) { console.log('I got called', value) }
```

image

**Review by Wadeck — APPROVED** [review]
Not manually tested but seems fine
Security approved

@yaroslavafenkin if you want to review it as you found the bug in the other PR ;)

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
